mewarpx/mewarpx/runinfo.py

```diff
@@ -0,0 +1,397 @@
+"""Methods for storing parameters from a run.
+
+This should all be made to Pickle easily, so that we can load the parameters at
+a later time.
+"""
+import collections
+import copy
+import datetime
+import logging
+import os
+import platform
+import subprocess
+
+import dill
+import mewarpx
+import numpy as np
+
+import mewarpx.util as mwxutil
+
+from mewarpx.mwxrun import mwxrun
+from functools import reduce
+
+logger = logging.getLogger(__name__)
+
+class SimInfo(object):
+
+    """Store basic simulation parameters used throughout analysis.
+
+    This must contain:
+        SimInfo.nxyz (int nx, int ny, int nz)
+        SimInfo.pos_lims (floats xmin, xmax, ymin, ymax, zmin, zmax)
+        SimInfo.geom (str 'XZ', 'RZ' or 'XYZ')
+        SimInfo.dt (float)
+        SimInfo.periodic (bool)
+
+    Arguments:
+        nxyz (int, int, int): tuple of the grid dimensions
+        pos_lims (floats xmin, xmax, ymin, ymax, zmin, zmax): tuple of the
+            position limits along each dimension
+        geom (str): geometry string 'XZ', 'RZ', or 'XYZ' for 2D grid,
+            cylindrical, or 3D grid
+        dt (float): simulation timestep interval
+        periodic (bool): use periodic boundary conditions
+
+    Note:
+        This base class has been created to provide additional plotting
+        functionality, and allow post-run creation of this object if needed.
+    """
+
+    def __init__(self, nxyz, pos_lims, geom, dt, periodic=True):
+        self.nxyz = nxyz
+        self.pos_lims = pos_lims
+        self.geom = geom
+        self.dt = dt
+        self.periodic = periodic
+
+    def get_vec(self, axis):
+        if axis == 'r':
+            return self.get_rvec()
+        axis_dict = {0: 0, 1: 1, 2: 2, 'x': 0, 'y': 1, 'z': 2}
+        axis = axis_dict[axis]
+        npts = self.nxyz[axis]
+        xmin = self.pos_lims[2*axis]
+        xmax = self.pos_lims[2*axis + 1]
+        # There is one more point on the grid than cell number
+        return np.linspace(xmin, xmax, npts + 1)
+
+    def get_rvec(self):
+        raise NotImplementedError
+        # npts = self.nxyz[0]
+        # xmin = 0.
+        # xmax = self.pos_lims[1]
+        # There is one more point on the grid than cell number
+        # return np.linspace(xmin, xmax, npts + 1)
+
+
+
+class WarpXSimInfo(SimInfo):
+
+    """Create a SimInfo object when running a warpx simulation."""
+
+    def __init__(self):
+        self.nxyz = [mwxrun.nx, mwxrun.ny, mwxrun.nz]
+        self.pos_lims = [mwxrun.xmin, mwxrun.xmax,
+                         mwxrun.ymin, mwxrun.ymax,
+                         mwxrun.zmin, mwxrun.zmax]
```

Comment:
The docstring for `SimInfo` says that nxyz and pos_lims should be tuples. The definition here should be consistent with the documentation of `SimInfo`, so they should both be tuples, or both be lists.